modules/pam_filter/pam_filter.c

```diff
@@ -128,92 +128,54 @@ static int process_args(pam_handle_t *pamh
 	}
 
 	strcpy(levp[0], ARGS_NAME);
-	size = ARGS_OFFSET;
 	for (i=0; i<argc; ++i) {
 	    if (i)
-		levp[0][size++] = ' ';
-	    strcpy(levp[0]+size, argv[i]);
-	    size += strlen(argv[i]);
+		strcat(levp[0], " ");
+	    strcat(levp[0], argv[i]);
```

Comment:
Potentially quadratic runtime behavior (for long strings).

Reply:
Adjusted to use stpcpy (POSIX 2008). Alternatively the commit could simply be removed.